python/cudf/cudf/core/series.py

```diff
@@ -1259,6 +1274,21 @@ def _getitem_preprocessed(
     def __getitem__(self, arg):
         if isinstance(arg, slice):
             return self.iloc[arg]
+        elif is_integer(arg) and not (
+            (
+                isinstance(self.index.dtype, CategoricalDtype)
+                and self.index.dtype.categories.dtype.kind in {"i", "u", "f"}
+            )
+            or self.index.dtype.kind in {"i", "u", "f"}
+        ):
+            warnings.warn(
+                "Series.__getitem__ treating keys as positions is deprecated "
+                "In a future version, integer keys will always be treated as labels "
```

Comment:
Do we know what version (is this us deprecating or pandas)? If we do and it's tied to a pandas version, can we add an explicit assertion above so that it fails once we hit that version?